jetsocat/src/tcp_proxy.rs

```diff
@@ -0,0 +1,186 @@
+use anyhow::{anyhow, Context as _, Result};
+use futures_util::FutureExt;
+use jet_proto::JET_VERSION_V2;
+use slog::{debug, o, Logger};
+use std::net::SocketAddr;
+use tokio::{net::TcpStream, pin};
+use uuid::Uuid;
+
+pub struct TcpProxyCmd {
+    pub source_addr: String,
+    pub association_id: String,
+    pub candidate_id: String,
+    pub auto_reconnect: bool,
+}
+
+struct TcpServer {
+    source_addr: SocketAddr,
+    jet_listener_addr: SocketAddr,
+    association_id: uuid::Uuid,
+    candidate_id: uuid::Uuid,
+    auto_reconnect: bool,
+}
+
+impl TcpServer {
+    pub fn new(
+        source_addr: SocketAddr,
+        jet_listener_addr: SocketAddr,
+        association_id: uuid::Uuid,
+        candidate_id: uuid::Uuid,
+        auto_reconnect: bool,
+    ) -> Self {
+        Self {
+            source_addr,
+            jet_listener_addr,
+            association_id,
+            candidate_id,
+            auto_reconnect,
+        }
+    }
+
+    pub async fn serve(self, log: Logger) -> Result<()> {
+        debug!(log, "Performing rendezvous connect...");
+
+        loop {
```

Comment:
@pacmancoder can you explain the reason for the loop? I believe it may be related to the problem where RDP makes multiple TCP connections during the negotiation step? Features like RDP auto-reconnect probably won't work without requesting a new JWT to open the connection, because the original JWT would be expired. The problem I could see is this might loop forever, we need a proper exit condition.

Reply:
Yes, the reasoning behind this is that mstsc reconnects during negotiation step. This happens once during the connection. Current jwt token validity interval, which is set from den is enough to perform reconnection. Maybe one improvement which we can make is to limit possible reconnection instead of loop-until-fail approach (the loop currently exits when Gateway kills association because of inactivity)

Reply:
Well, in the case of RDP this code is indeed required to make RDP forwarding function normally. However, in general, for general tcp-forwarding, I think this is optional. Let's pass auto-reconnect retries  from CLI (--max-reconnects=3), what do you think?

Reply:
Indeed I prefer the `--max-reconnects` approach!

Reply:
@pacmancoder I discussed with Marc-André and we agreed on renaming to `--max-reconnection-count` with a default value of 0 (only initial connection).
Loop would be:
```rust
for _ in 0..=2 {
    …
}
```
(notice the ..**=**)